File: account/src/main/java/serp/project/account/core/domain/dto/request/CreatePermissionDto.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */


package serp.project.account.core.domain.dto.request;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class CreatePermissionDto {
    @NotBlank
    private String name;
    private String description;
}

File: api_gateway/src/core/domain/dto/request/account/create_permission_dto.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package request

type CreatePermissionDto struct {
	Name        string  `json:"name"`
	Description *string `json:"description"`
	Resource    string  `json:"resource"`
	Action      string  `json:"action"`
}
